File: algorithms/java/src/ReverseInteger.java
/**
 * @author jasper
 * @create 2021-09-26 18:14
 */
public class ReverseInteger {
    public int reverse(int x) {
        int rev = 0;
        while (x != 0) {
            int pop = x % 10;
            x /= 10;
            if (rev > Integer.MAX_VALUE / 10 || rev < Integer.MIN_VALUE / 10){
                return 0;
            }
            rev = rev * 10 + pop;
        }
        return rev;
    }
}

File: algorithms/python/ReverseInteger.py
# @author jasper
# @create 2021-09-26 18:15

class Solution:
    def reverse(self, x: int) -> int:
        flag = False
        if x < 0:
            flag = True
            x = -x
        l = [i for i in str(x)]
        l.reverse()
        n = int("".join(l))
        n = -n if flag else n
        n = 0 if n < -2 ** 31 or n > 2 ** 31 - 1 else n
        return n
